x-pack/plugins/observability_solution/apm/public/components/shared/transaction_action_menu/sections.ts

```diff
@@ -117,13 +119,14 @@ export const getSections = ({
       label: i18n.translate('xpack.apm.transactionActionMenu.showPodMetricsLinkLabel', {
         defaultMessage: 'Pod metrics',
       }),
-      href: getInfraHref({
-        app: 'metrics',
-        basePath,
-        path: `/link-to/pod-detail/${podId}`,
-        query: infraMetricsQuery,
+      href: assetDetailsLocator?.getRedirectUrl({
+        assetId: podId!,
```

Comment:
should we even build the link if `podId`, `containerId` and `hostName` are undefined? thinking here if we could avoid using `!`

Reply:
If I understand correctly, there's the `condition` property that would apply whether the link gets generated or not? If it is false, I assume from the reading that the link won't generate. For the `true` condition, we then know there will be a `podId`, etc, hence the `!`.